src/routes/boot.ts

```diff
@@ -328,6 +331,48 @@ const getUser = (con: DataSource, userId: string): Promise<User> =>
     ],
   });
 
+const getMarketingCta = async (con: DataSource, userId: string) => {
+  let marketingCta: MarketingCta | null = null;
+  const redisKey = generateStorageKey(
+    StorageTopic.Boot,
+    'marketing_cta',
+    userId,
+  );
+  const rawRedisValue = await getRedisObject(redisKey);
+
+  if (rawRedisValue === RedisMagicValues.SLEEPING) {
+    // TODO: remove log or replace with proper logging
+    console.log('sleeping, not fetching from db');
+    return null;
+  } else {
+    marketingCta = JSON.parse(rawRedisValue);
+  }
+
+  if (!marketingCta) {
+    // TODO: remove log or replace with proper logging
+    console.log('fetching from db');
+    const userMarketingCta = await con.getRepository(UserMarketingCta).findOne({
+      where: { userId, readAt: IsNull() },
+      order: { createdAt: 'ASC' },
+      relations: ['marketingCta'],
+    });
+
+    if (userMarketingCta) {
+      marketingCta = userMarketingCta.marketingCta;
+      await setRedisObject(redisKey, JSON.stringify(marketingCta));
+    } else {
+      marketingCta = null;
+      // TODO: replace 10 with ONE_HOUR_IN_SECONDS
+      await setRedisObjectWithExpiry(redisKey, RedisMagicValues.SLEEPING, 10);
+    }
+  } else {
+    // TODO: remove log or replace with proper logging
+    console.log('fetched from redis');
```

Comment:
Probably the whole else can be removed.

Reply:
Will remove, but keeping it for now as debugging